Assets/_GameJamCat/Scripts/CatGenerator.cs

```diff
@@ -0,0 +1,159 @@
+        private readonly int NoiseScaleID = Shader.PropertyToID("_NoiseScale");
+        #endregion
+
+
+        [SerializeField]
+        CatBehaviour _catPrefab;
+        [SerializeField]
+        Texture2D[] _furs;
+        [SerializeField, ColorUsage(false)]
+        Color[] _furColors, _feetColors, _earColors;
+        [SerializeField, ColorUsage(false, true)]
+        Color[] _eyeColors;
+
+        [SerializeField, Range(0f, 1f), Tooltip("Percent of cats with textures, relative to solid colors")]
+        float _percentTextured = 0.5f;
+        [SerializeField, MinMaxSlider(-10.0f, 10.0f)]
+        Vector2 _textureOffsetRange = new Vector2(-5f, 5f);
+        [SerializeField, MinMaxSlider(0.0f, 5.0f)]
+        Vector2 _textureScaleRange = new Vector2(1f, 5f);
+        [SerializeField, MinMaxSlider(0.0f, 1.0f)]
+        Vector2 _noiseWeightRange = new Vector2(0f, 1f);
+        [SerializeField, MinMaxSlider(0.0f, 10f)]
+        Vector2 _noiseScaleRange = new Vector2(0f, 5f);
+
+        [SerializeField]
+        bool _poolObjects = false;
+        [System.NonSerialized]
+        List<CatBehaviour> _pool = null;
+        Scene _poolScene;
+
+        public CatBehaviour GetRandomCat()
+        {
+            var newCat =  _poolObjects ? GetRandomCat_pooled() : GetRandomCat_unpooled();
+            SetupCatMaterial(newCat.CatRenderer.material);
+            
+            return newCat;
+        }
+
+        public void Reclaim(CatBehaviour cat)
+        {
+            if (!_poolObjects)
+            {
+                Destroy(cat.gameObject);
+                return;
+            }
+            if (_pool == null) CreatePool();
+            cat.gameObject.SetActive(false);
+            _pool.Add(cat);
+        }
+
+        private void SetupCatMaterial(Material m)
+        {
+            if(Random.value <= _percentTextured)
+            {
+                //Texture the cat
+                // Todo: it would be neat to assign some color variance alongside the texture
+                m.SetTexture(FurMapID, GetRandom(_furs));
+                m.SetVector(FurOffsetID, new Vector2(GetRandom(_textureOffsetRange), GetRandom(_textureOffsetRange)));
+                m.SetVector(FurScaleID, new Vector2(GetRandom(_textureScaleRange), GetRandom(_textureScaleRange)));
+                m.SetColor(FurColorID, Color.white);
+            }
+            else
+            {
+                //Leave texture null and assign a color
+                //m.SetTex
+                m.SetColor(FurColorID, GetRandom(_furColors));
+            }
+            m.SetColor(FeetColorID, GetRandom(_feetColors));
+            m.SetColor(EarColorID, GetRandom(_earColors));
+            m.SetColor(EyeColorID, GetRandom(_eyeColors));
+            m.SetFloat(NoiseScaleID, GetRandom(_noiseScaleRange));
+            m.SetFloat(NoiseWeightID, GetRandom(_noiseWeightRange));
+        }
+
+        private CatBehaviour GetRandomCat_pooled()
```

Comment:
Should stick with pascal case and not introduce snake case in middle. 
So you can write 
GetPooledRandomCat or something

same with 
GetRandomCat_unpooled -> GetUnpooledRandomCat